spec/patterns/core_spec.rb

```diff
@@ -114,6 +114,83 @@
       expect(grok_match(pattern,value)).to pass
     end
   end
+
+  context "dotted path" do
+
+    it "should match path containing ." do
+      expect(grok_match(pattern, '/some/./path/')).to pass
+      expect(grok_match(pattern, '/some/../path')).to pass
+      expect(grok_match(pattern, '/../.')).to pass
+      expect(grok_match(pattern, '/.')).to pass
+      expect(grok_match(pattern, '/..')).to pass
+      expect(grok_match(pattern, '/...')).to pass
+
+      expect(grok_match(pattern, 'a/./b/c')).to pass
+      expect(grok_match(pattern, ',/.')).to pass
+      expect(grok_match(pattern, '+/.../')).to pass
+    end
+
+    it "should match path starting with ." do
+      expect(grok_match(pattern, '../0')).to pass
+      expect(grok_match(pattern, './~')).to pass
+      expect(grok_match(pattern, '.../-')).to pass
+      expect(grok_match(pattern, './')).to pass
+      expect(grok_match(pattern, './,')).to pass
+      expect(grok_match(pattern, '../')).to pass
+      expect(grok_match(pattern, '.a/')).to pass
+      expect(grok_match(pattern, '.~/')).to pass
+    end
+
+    it "should not match if there's no separator" do
+      expect(grok_match(pattern, '.')).to_not pass
+      expect(grok_match(pattern, '..')).to_not pass
+      expect(grok_match(pattern, '...')).to_not pass
+      expect(grok_match(pattern, '.,')).to_not pass
+      expect(grok_match(pattern, '.-')).to_not pass
+    end
+
+  end
+
+  context "separators" do
+
+    it "should match root" do
+      expect(grok_match(pattern, '/')).to pass
+    end
+
+    it "should match" do # NOTE: we did not match these in < 4.2.0
+      expect(grok_match(pattern, '//')).to pass
```

Comment:
multiple slashes `//` did not match using the old pattern so this could be considered "breaking" but I would incline towards it rather being a bug as `//` is allowed according to the Unix path-name specification. 
there's a [good summary](https://unix.stackexchange.com/questions/256497/on-what-systems-is-foo-bar-different-from-foo-bar) of (mostly legacy) OSs these are not valid filesystem paths, for the curious.